### PR TITLE
update generated artifacts for k8s and openshift so that env variables are loaded in a particular order

### PR DESCRIPTION
**Pull request by procrypt**
Fixes: #595 
cc: @surajssd

### CODE REVIEW

**Comment by cdrage** [comment]
Although this sucks in terms of order, having ${FOOBAR} would be against what's promised in the spec file :( https://docs.docker.com/compose/compose-file/#args having $FOOBAR would help in terms of testing to *make sure* environment variables work.

**Comment by surajnarwade** [comment]
@cdrage , I didn't get what you said

**Comment by cdrage** [comment]
@surajnarwade 

@procrypt Changed the variables to ${FOOBAR} from $FOOBAR. They should stay $FOOBAR

**Comment by surajssd** [comment]
@procrypt yes that should not make any difference, keeping it without brackets `{` or `}` should be okay.

**Comment by procrypt** [comment]
@surajssd Updated.

File: script/test/fixtures/multiple-compose-files/output-k8s.json [hunk 1]
```diff
@@ -89,31 +79,28 @@
...
-                  {
-                    "containerPort": 9001
                   }
                 ],
                 "env": [
                   {
-                    "name": "DB_DBID",
-                    "value": "openshift"
+                    "name": "DB_HOST",
+                    "value": "kubernetes"
```
Comment (surajssd) [inline script/test/fixtures/multiple-compose-files/output-k8s.json]: The value should remain the same right? I mean the value should still be `openshift` right?

File: script/test/fixtures/multiple-compose-files/output-openshift.json [hunk 2]
```diff
@@ -255,19 +275,35 @@
                 ],
                 "env": [
                   {
+                    "name": "MYSQL_ROOT_PASSWORD",
+                    "value": "kubernetes"
```
Comment (surajssd) [inline script/test/fixtures/multiple-compose-files/output-openshift.json]: Is this config correct? Because same env is created twice? Will this run on a k8s cluster?

Seems like something libcompose is not handling?

File: script/test/fixtures/etherpad/docker-compose.yml [hunk 3]
```diff
@@ -6,10 +6,10 @@ services:
     ports:
       - "$DB_PORT"
     environment:
-      MYSQL_ROOT_PASSWORD: $ROOT_PASS
-      MYSQL_DATABASE: $DB_NAME
-      MYSQL_PASSWORD: $DB_PASS
-      MYSQL_USER: $DB_USER
+      - "MYSQL_ROOT_PASSWORD=$ROOT_PASS"
```
Comment (cdrage) [inline script/test/fixtures/etherpad/docker-compose.yml]: You're still changing these values!

Leave them be please, no quotes, no editing them. They should stay the same as before.

File: script/test/fixtures/restart-options/docker-compose-restart-no.yml [hunk 4]
```diff
@@ -5,5 +5,5 @@ services:
     image: "foobar"
     restart: "no"
     environment:
-      GITHUB: surajssd
+      - "GITHUB=surajssd"
```
Comment (cdrage) [inline script/test/fixtures/restart-options/docker-compose-restart-no.yml]: Again.. quotes everywhere. Leave them the same as before.

**Comment by procrypt** [comment]
Ping @surajssd @cdrage @surajnarwade need review.

**Comment by kadel** [comment]
This means that we are no longer testing doceker-compose files with evn variables in map?

**Comment by cdrage** [comment]
Tests are failing :( Isn't this PR only re-organizing the env variables in the tests? I don't believe it's removing them? @kadel

**Comment by kadel** [comment]
If I'm looking correctly this PR changes env vars. in all docker-compose files from map to array no?

**Comment by procrypt** [comment]
@kadel Yes.

**Comment by procrypt** [comment]
@kadel Some file [1](https://github.com/kubernetes-incubator/kompose/blob/master/script/test/fixtures/restart-options/docker-compose-restart-no.yml)  and [2](https://github.com/kubernetes-incubator/kompose/blob/master/script/test/fixtures/restart-options/docker-compose-restart-onfail.yml) has not been changed.

File: script/test/fixtures/etherpad/docker-compose.yml [hunk 5]
```diff
@@ -6,10 +6,10 @@ services:
     ports:
       - "$DB_PORT"
     environment:
-      MYSQL_ROOT_PASSWORD: $ROOT_PASS
-      MYSQL_DATABASE: $DB_NAME
-      MYSQL_PASSWORD: $DB_PASS
-      MYSQL_USER: $DB_USER
```
Comment (cdrage) [inline script/test/fixtures/etherpad/docker-compose.yml]: Sstop changing these. Just leave them as they are with the `:`. No need to change ANY of them.

Comment (kadel) [inline script/test/fixtures/etherpad/docker-compose.yml]: :+1: docker-compose files should stay same as they were

**Comment by procrypt** [comment]
ping @cdrage @surajssd updated PR

**Comment by cdrage** [comment]
LGTM

File: script/test/fixtures/keyonly-envs/env.yml [hunk 6]
```diff
@@ -10,10 +10,10 @@ services:
     ports:
       - "6379"
     environment:
-      - GET_HOSTS_FROM=dns
-      - RACK_ENV=development
-      - SHOW=true
-      - SESSION_SECRET
+      GET_HOSTS_FROM: dns
```
Comment (kadel) [inline script/test/fixtures/keyonly-envs/env.yml]: why is this changed from list to map?

**Comment by procrypt** [comment]
@kadel I might have dome it by mistake, updated the PR.

**Comment by surajssd** [comment]
@procrypt this change is done on basis of what code now? Also how is this sorting being done? Also if this sorting is done on basis of `env`'s `name` then I see bunch of them not sorted!

screenshot from 2017-06-14 12-20-53


See the screenshot above, like `DB_HOST` is before `DB_DBID` which wrong sorting!

**Comment by procrypt** [comment]
@surajssd I'm checking it.

**Comment by procrypt** [comment]
@surajssd once #631 gets merged this issue will be solved. I have updated #631 and it already have 2 LGTM can you PTAL and merge it soon.

**Comment by procrypt** [comment]
@surajssd Ping

**Comment by surajssd** [comment]
@procrypt please rebase on current master, lot of new changes have been added since morning, and also update the commit message since we are just updating the fixtures/configs and not changing the docker-compose!

For me lot of tests fail, not sure why!

**Comment by procrypt** [comment]
@surajssd rebased

**Comment by surajssd** [comment]
@procrypt tests pass for me, please update the commit message since we are not changing docker-compose file?

**Comment by procrypt** [comment]
@surajssd Done.

**Comment by surajssd** [comment]
@procrypt but seems like it needs rebase now

**Comment by procrypt** [comment]
@surajssd Rebased.

**Comment by surajnarwade** [comment]
LGTM :)
